client/lib/state_stack.js

```diff
@@ -70,7 +70,14 @@ class StateStack {
 
 			// Display
 			this.clear()
-			invoke(this.state, "display")
+			if (stateName){
+				if (stateName === this.state.stateName){
```

Comment:
This worked because undefined === undefined, so it was really only skipping the first call